core/trino-main/src/main/java/io/trino/operator/PartitionFunctionFactory.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.operator;
+
+import com.google.common.primitives.Ints;
+import io.airlift.slice.XxHash64;
+import io.trino.Session;
+import io.trino.operator.exchange.LocalPartitionGenerator;
+import io.trino.spi.type.Type;
+import io.trino.sql.planner.MergePartitioningHandle;
+import io.trino.sql.planner.NodePartitioningManager;
+import io.trino.sql.planner.PartitioningHandle;
+import io.trino.sql.planner.SystemPartitioningHandle;
+import io.trino.type.BlockTypeOperators;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class PartitionFunctionFactory
+{
+    private final NodePartitioningManager nodePartitioningManager;
+    private final BlockTypeOperators blockTypeOperators;
+
+    public PartitionFunctionFactory(NodePartitioningManager nodePartitioningManager, BlockTypeOperators blockTypeOperators)
+    {
+        this.nodePartitioningManager = requireNonNull(nodePartitioningManager, "nodePartitioningManager is null");
+        this.blockTypeOperators = requireNonNull(blockTypeOperators, "blockTypeOperators is null");
+    }
+
+    public PartitionFunction create(
```

Comment:
this is just refactor, right?

Reply:
This is required because we are using this in both `LocalExchange` as well as `PartitionTableWriterOperator`